kv: TestGossipRestart triggers checksum death

`rm -rf acceptance/boo/; while [ $? -eq 0 ]; do make acceptance TESTS=GossipRes TESTTIMEOUT=20m TESTFLAGS='-v -nodes 5 -count 1 -l boo'; done`

has nodes crashing with

```
I160310 05:22:18.714630 gossip/gossip.go:938  starting client to roach2:26257
W160310 05:22:19.772736 storage/store.go:802  [node=1,store=1] error gossiping system config: range 0: replica <nil> not leader; leader is <nil>
panic: checksums: e = [175 213 109 217 231 197 231 128 143 190 153 189 164 172 131 145 254 138 220 144 171 242 228 137 133 183 63 29 72 77 16 143 160 163 180 131 33 112 130 40 169 188 144 205 211 205 96 60 148 238 135 149 69 163 98 71 220 236 46 74 134 235 140 198], v = [107 203 119 224 213 167 178 74 130 192 76 110 233 69 133 208 42 254 245 237 193 53 185 126 232 77 233 26 231 128 160 182 73 72 97 185 80 170 22 148 201 84 121 75 36 115 216 137 208 133 177 161 24 65 222 223 147 11 245 200 124 244 8 179]

goroutine 16 [running]:
panic(0x13e32c0, 0xc8206052d0)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/storage.(*Replica).VerifyChecksum(0xc8200c4380, 0x7ff8b6129608, 0xc8206c64b0, 0xc8205fd020, 0x143a63f68549389e, 0x0, 0x0, 0x0, 0x3, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica_command.go:1496 +0x829
github.com/cockroachdb/cockroach/storage.(*Replica).executeCmd(0xc8200c4380, 0x7ff8b6129608, 0xc8206c64b0, 0xc8205fd020, 0x143a63f68549389e, 0x0, 0x0, 0x0, 0x3, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica_command.go:160 +0x1bd2
github.com/cockroachdb/cockroach/storage.(*Replica).executeBatch(0xc8200c4380, 0x7ff8b6129608, 0xc8206c64b0, 0xc8205fd020, 0x143a63f68549389e, 0x0, 0x0, 0x0, 0x3, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1605 +0x3e0
github.com/cockroachdb/cockroach/storage.(*Replica).applyRaftCommandInBatch(0xc8200c4380, 0x7ff8b614f520, 0xc8206c6450, 0x2c, 0x200000002, 0xc800000002, 0x143a63f68549389e, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1468 +0x433
github.com/cockroachdb/cockroach/storage.(*Replica).applyRaftCommand(0xc8200c4380, 0x7ff8b614f520, 0xc8206c6450, 0x2c, 0x200000002, 0xc800000002, 0x143a63f68549389e, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1370 +0x490
github.com/cockroachdb/cockroach/storage.(*Replica).processRaftCommand(0xc8200c4380, 0xc820605198, 0x8, 0x2c, 0x3, 0x200000002, 0x2, 0x143a63f68549389e, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1336 +0x308
github.com/cockroachdb/cockroach/storage.(*Replica).handleRaftReady(0xc8200c4380, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1197 +0xa53
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1872 +0x35a
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8202dd490, 0xc8202002c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62
```

--- a/acceptance/chaos_test.go
+++ b/acceptance/chaos_test.go
@@ -418,8 +418,7 @@ func testNodeRestartInner(t *testing.T, c cluster.Cluster, cfg cluster.TestConfi
 	log.Infof("%d %.1f/sec", count, float64(count)/elapsed.Seconds())
 	kvClient, kvStopper := c.NewClient(t, num-1)
 	defer kvStopper.Stop()
-	if pErr := kvClient.CheckConsistency(keys.TableDataMin, keys.TableDataMax); pErr != nil {
-		// TODO(.*): change back to t.Fatal after #5051.
-		log.Error(pErr)
+	if err := kvClient.CheckConsistency(keys.TableDataMin, keys.TableDataMax); err != nil {
+		log.Fatal(err)
 	}
 }

--- a/storage/replica_command.go
+++ b/storage/replica_command.go
@@ -1532,8 +1532,7 @@ func (r *Replica) VerifyChecksum(batch engine.Engine, ms *engine.MVCCStats, h ro
 		if p := r.store.ctx.TestingKnobs.BadChecksumPanic; p != nil {
 			p()
 		} else {
-			// TODO(.*): see #5051.
-			log.Errorf("checksum mismatch: e = %x, v = %x", args.Checksum, c.checksum)
+			panic(fmt.Sprintf("checksum mismatch: e = %x, v = %x", args.Checksum, c.checksum))
 		}
 	}
 	return roachpb.VerifyChecksumResponse{}, nil

